clientv3/balancer/balancer_test.go

```diff
@@ -92,24 +91,32 @@ func TestRoundRobinBalancedResolvableNoFailover(t *testing.T) {
 				return picked, err
 			}
 
-			prev, switches := "", 0
+			var picked string
+			available := make(map[string]struct{})
+			// cycle through all peers to indirectly verify that balancer subconn list is fully loaded
+			// otherwise we can't reliably count switches in the next step
+			for len(available) < tc.serverCount {
```

Comment:
FYI, technically we only need 2 different peers for switch counting logic to work